src/oidc/error.ts

```diff
@@ -0,0 +1,34 @@
+export enum OIDCErrorType {
+    FailedToFetchOIDCConfiguration = "FailedToFetchOIDCConfiguration",
+    AuthenticationRequestFailed = "AuthenticationRequestFailed",
+    AccessTokenRequestFailed = "AccessTokenRequestFailed",
+    LegacyTokenRequestFailed = "LegacyTokenRequestFailed",
+    UserManagerCreationFailed = "UserManagerCreationFailed"
```

Comment:
I didn't use integers because I wanted to pass this error type into the `Error` interface, which does not have any integer attribute supported 
<img width="251" alt="Screenshot 2024-11-06 at 2 58 17 PM" src="https://github.com/user-attachments/assets/30fc6902-7000-45fc-a217-da85359384a7">